03-intro-to-python.Rmd

```diff
@@ -1,9 +1,298 @@
+
+When you want to document your code, you can use Jupyter Notebooks. Jupyter Notebooks are an open source web application for easily sharing documents that contain your live Python code, equations, visualizations and data science explanations. 
+
+Python is particularly suited for large scale machine learning and deep learning with libraries such as TensorFlow, PyTorch, and scikit-learn.
+
 
 ## Learning Objectives
 
 ```{r, fig.align='center', echo = FALSE, fig.alt= "Major point!! example image"}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1k8uC1rqnGTSbKjBsWvKYgiUUxO1q_VhJCwZQHJNWozA/edit#slide=id.g29054a882fd_0_52")
 ```
+
+## Python Syntax for R Users
+
+Most important difference in syntax is 0-based indexing for Python and 1-based indexing for R. This means that in R, indexing starts with 1 and in Python, indexing starts with 0. Coming from R, this means you have to subtract your "R indexes" by 1 to get the correct index in Python.
+
+Other major differences in Python:
+
+### Whitespace
+
+Important in Python. In R, expressions are grouped into a code block with `{}`. In Python, expressions are grouped by indentation level. 
+
+For example, in R, an if statement looks like:
+
+```{r}
+x <- 1
+
+if (x > 0) {
+    print("x is positive")
+} else {
+    print("x is negative")
+}
+```
+
+In Python, the equivalent if statement looks like:
+
+```{python}
+x = 1
+
+if x > 0:
+    print("x is positive")
+else:
+    print("x is negative")
+```
+
+
+### Data Structures
+
+There are 4 different data storage formats, or data structures, in Python: lists, tuples, dictionaries, and sets
+
+#### Lists
+
+Python lists are created using brackets `[]`. You can add elements to the list through the `append()` method.
+
+```{python}
+x = [1, 2, 3]
+x.append(4) # add 4 to the end of list
+
+print("x is", x)
+#> x is [1, 2, 3, 4]
+```
+
+
+You can index into lists with integers using brackets `[]`, but note that indexing is 0-based.
+
+```{python}
+x = [1, 2, 3]
+
+x[0]
+#> 1
+x[1]
+#> 2
+x[2]
+#> 3
+```
+
+
+Negative numbers count from the end of the list.
+
+```{python}
+x = [1, 2, 3]
+
+x[-1]
+#> 3
+x[-2]
+#> 2
+x[-3]
+#> 1
+```
+
+You can slice ranges of lists using the : inside brackets. Note that the slice syntax is not inclusive of the end of the slice range.
+
+```{python}
+x = [1, 2, 3, 4, 5, 6]
+x[0:2] # get items at index positions 0, 1
+#> [1, 2]
+x[1:]  # get items from index position 1 to the end
+#> [2, 3, 4, 5, 6]
+x[:-2] # get items from beginning up to the 2nd to last.
+#> [1, 2, 3, 4]
+x[:]   # get all the items
+#> [1, 2, 3, 4, 5, 6]
+```
+
+
+#### Tuples
+
+Tuples behave like lists, but are constructued using `()`, instead of `[]`. 
```

Comment:
Minor typo here in "constructed." Another important difference is that tuples are immutable.